Code/Framework/AzToolsFramework/AzToolsFramework/Asset/AssetSeedManager.cpp

```diff
@@ -772,11 +772,42 @@ namespace AzToolsFramework
         }
 
         AzFramework::AssetSeedList assetSeedList;
-        if (!AZ::Utils::LoadObjectFromFileInPlace(sourceFilePath.c_str(), assetSeedList))
+
+        // As the seed file can support both JSON and ObjectStream XML
+        // It is opened and read into memory first
+        AZ::IO::FileIOStream assetSeedStream;
+        if (!assetSeedStream.Open(sourceFilePath.c_str(), AZ::IO::OpenMode::ModeRead | AZ::IO::OpenMode::ModeBinary))
         {
             return false;
         }
 
+        auto ReadAssetSeedData = [&assetSeedStream](char* buffer, size_t size) -> size_t
+        {
+            return assetSeedStream.Read(size, buffer);
+        };
+
+        AZStd::string assetSeedData;
+        assetSeedData.resize_and_overwrite(assetSeedStream.GetLength(), ReadAssetSeedData);
+
+        // If the asset seed file starts with the ObjectStream XML stream tag
+        // then load the data using the ObjectStream
+        constexpr AZ::u8 xmlObjectStreamTag = '<';
+        if (assetSeedData.starts_with(xmlObjectStreamTag))
+        {
+            if (!AZ::Utils::LoadObjectFromBufferInPlace(assetSeedData.c_str(), assetSeedData.size(), assetSeedList))
+            {
+                return false;
+            }
+        }
+        else
+        {
+            // Load the asset seed file using Json Serialization
+            if (!AZ::JsonSerializationUtils::LoadObjectFromString(assetSeedList, assetSeedData))
+            {
+                return false;
+            }
+        }
```

Comment:
This is the updated Asset Manager Loading logic.

It can seen here that the code will still load the ObjectStream XML formatted files and the change is backward compatible.